guides/source/action_cable_overview.md

```diff
@@ -256,24 +260,35 @@ When a consumer is subscribed to a channel, they act as a subscriber. This
 connection is called a subscription. Incoming messages are then routed to
 these channel subscriptions based on an identifier sent by the cable consumer.
 
-```coffeescript
-# app/assets/javascripts/cable/subscriptions/chat.coffee
-# Assumes you've already requested the right to send web notifications
-App.cable.subscriptions.create { channel: "ChatChannel", room: "Best Room" },
-  received: (data) ->
-    @appendLine(data)
-
-  appendLine: (data) ->
-    html = @createLine(data)
-    $("[data-chat-room='Best Room']").append(html)
-
-  createLine: (data) ->
-    """
-    <article class="chat-line">
-      <span class="speaker">#{data["sent_by"]}</span>
-      <span class="body">#{data["body"]}</span>
-    </article>
-    """
+```js
+// app/javascript/cable/subscriptions/chat_channel.js
+// Assumes you've already requested the right to send web notifications
+import consumer from "./consumer"
+
+consumer.subscriptions.create(
+  { channel: "ChatChannel", room: "Best Room" },
+  {
+    received: function(data) {
+      this.appendLine(data)
+    },
+
+    appendLine: function(data) {
+      html = this.createLine(data)
+      $("[data-chat-room='Best Room']").append(html)
+    },
+
+    createLine: function(data) {
+      return (
+        `
+        <article class="chat-line">
+          <span class="speaker">${data["sent_by"]}</span>
+          <span class="body">${data["body"]}</span>
+        </article>
+        `
+      )
+    }
+  }
+)
```

Comment:
Here's how I'd write the above example using my feedback above and adjusting the formatting a bit:
```js
consumer.subscriptions.create({ channel: "ChatChannel", room: "Best Room" }, {
  received(data) {
    this.appendLine(data)
  },

  appendLine(data) {
    const html = this.createLine(data)
    const element = document.querySelector("[data-chat-room='Best Room']")
    element.insertAdjacentHTML("beforeend", html)
  },

  createLine(data) {
    return `
      <article class="chat-line">
        <span class="speaker">${data["sent_by"]}</span>
        <span class="body">${data["body"]}</span>
      </article>
    `
  }
})
```